lib/host/host.go

```diff
@@ -575,14 +578,44 @@ func getHostDirs() []string {
 
 var hostMap map[string]*Host
 
+func intiHostWorker(hosts chan string, res chan *Host) {
+	for hostDir := range hosts {
+		res <- InitHost(hostDir)
+	}
+}
+
+var hostMapMutex sync.Mutex
+
 func getHosts() map[string]*Host {
+	//defer profiler.Timer("getHosts")()
 	if hostMap != nil {
 		return hostMap
 	}
 
-	for _, hostDir := range getHostDirs() {
-		hostMap[hostDir] = InitHost(hostDir)
+	workers := make(chan string, 20)
+	c := make(chan *Host)
+	for i := 0; i < cap(workers); i++ {
+		go intiHostWorker(workers, c)
+	}
+
+	hostDirs := getHostDirs()
+	go func() {
+		for _, host := range hostDirs {
+			workers <- host
+		}
+	}()
+
+	for i := 0; i < len(hostDirs); i++ {
+		host := <-c
+
+		hostMapMutex.Lock()
```

Comment:
I don't think we need to have a mutex when the main thread is the only thread adding anything to the host map